src/hydrabus/hydrabus_mode_onewire.c

```diff
@@ -180,58 +215,149 @@ uint8_t onewire_read_u8(t_hydra_console *con)
 	return value;
 }
 
-void onewire_scan(t_hydra_console *con)
+static uint8_t onewire_crc8(uint8_t value, struct onewire_scan_state *state)
 {
-	uint8_t id_bit_number = 0;
-	uint8_t last_zero = 0;
-	uint8_t id_bit = 0, cmp_id_bit = 0;
-	uint8_t search_direction = 0;
-	uint8_t LastDiscrepancy = 0;
-	uint8_t LastDeviceFlag = 0;
-	uint8_t i;
-	uint8_t ROM_NO[8] = {0};
-	onewire_start(con);
-	onewire_write_u8(con, 0xf0);
-	cprintf(con, "Discovered devices : ");
-	while(!LastDeviceFlag && !hydrabus_ubtn()) {
-		do{
+	state->crc8 = onewire_crc_table[state->crc8 ^ value];
+
+	return state->crc8;
+}
+
+static bool onewire_search(t_hydra_console *con, struct onewire_scan_state *state, enum onewire_scan_mode mode)
+{
+	int id_bit_number;
+	int last_zero, rom_byte_number, search_result;
```

Comment:
declare `search_result` as `bool` to match the function return type

Reply:
Fixed.